lib/beaker-pe/pe-client-tools/executable_helper.rb

```diff
@@ -0,0 +1,153 @@
+        # @param [BEAKER::Host] host The SUT that should run the puppet-app command
+        # @param [String] args The arguments to puppet-app
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_app_on(*args, &block)
+          Private.new.tool(:app, *args, &block)
+        end
+
+        # puppet-db helper win/lin/osx
+        # @param [BEAKER::Host] host The SUT that should run the puppet-db command
+        # @param [String] args The arguments to puppet-db
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_db_on(*args, &block)
+          Private.new.tool(:db, *args, &block)
+        end
+
+        # puppet-query helper win/lin/osx
+        # @param [BEAKER::Host] host The SUT that should run the puppet-query command
+        # @param [String] args The arguments to puppet-query
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_query_on(*args, &block)
+          Private.new.tool(:query, *args, &block)
+        end
+
+        # Logs a user in on a SUT with puppet-access/RBAC API (windows)
+        # @param [Beaker::Host] host The SUT to perform the login on
+        # @param [Scooter::HttpDispatchers::ConsoleDispatcher] credentialed_dispatcher A Scooter dispatcher that has credentials for the user
+        # @option attribute_hash [String] :name The environment variable
+        # @option attribute_hash [String] :default The default value for the environment variable
+        # @option attribute_hash [String] :message A message describing the use of this variable
+        # @option attribute_hash [Boolean] :required Used internally by CommandFlag, ignored for a standalone EnvVar
+        def login_with_puppet_access_on(host, credentialed_dispatcher, opts={})
+
+          lifetime = opts[:lifetime] || nil
+          unless host.platform =~ /win/
+
+            user = credentialed_dispatcher.credentials.login
+            password = credentialed_dispatcher.credentials.password
+            puppet_access_on(host, 'login', {:stdin => "#{user}\n#{password}\n"})
+          else
+
+            # this is a hack
+            # puppet-access needs to support alternative to interactive login
+            # create .puppetlabs dir
+            cmd = Beaker::Command.new('echo', ['%userprofile%'], :cmdexe => true)
+            user_home_dir = host.exec(cmd).stdout.chomp
+            win_token_path =  "#{user_home_dir}\\.puppetlabs\\"
+            host.exec(Beaker::Command.new('MD', [win_token_path.gsub('\\', '\\\\\\')], :cmdexe => true), :accept_all_exit_codes => true)
+
+            token = credentialed_dispatcher.acquire_token_with_credentials(lifetime)
+            create_remote_file(host, "#{win_token_path}\\token", token)
+          end
+        end
+
+        class Private
```

Comment:
why is this class named `Private`?
if you really want the class to only be used locally, it probably doesn't need to be a class?  use private methods
since it's all in this module. 

but you are using it to create commands above.  i'm with kevin on this.
inherit Command and overload #cmd_line ?